### ISSUE TITLE
wrong number of arguments (given 5, expected 4)

### ISSUE THREAD

**Issue by amalkumars** [iHiD/propono#105]
I am getting an Argument Error when using trying to publish a message. I am using propono 3.0, ruby-3 and rails-6

`[GEM_ROOT]/gems/propono-3.0.0/lib/propono/services/publisher.rb:14:in initialize' initialize: wrong number of arguments (given 5, expected 4) (ArgumentError)`
`[GEM_ROOT]/gems/propono-3.0.0/lib/propono/services/publisher.rb:9:in new'`
`[GEM_ROOT]/gems/propono-3.0.0/lib/propono/services/publisher.rb:9:in publish'`
`[GEM_ROOT]/gems/propono-3.0.0/lib/propono/components/client.rb:54:in publish'`

Are we having the support of Ruby3?

**Comment by iHiD** [comment]
Yeah - so I think Ruby 3 needs `**kwargs` wherever it uses `*args`. I don't have time to look at this until Exercism has launched (http://countdown.exercism.io/) but will after that. I'll happily accept a PR too. You can turn on Ruby 3 in CI here: https://github.com/iHiD/propono/blob/main/.github/workflows/tests.yml#L18

**Comment by BiggerNoise** [comment]
Hello

This one is giving us grief, so I'm working on a P/R for you.

I am running into one issue: Ruby 2.7 seems to be happy with the double splat changes in 3.0, but 2.6 seems to be choking on them.  (https://github.com/KoanHealth/propono/actions)

Unfortunately, I have an M1 mac and rbenv won't install anything earlier than 2.7, so I'm having trouble looking more closely at this.

If 2.6 support is key for y'all, then I'll see if I can rope a coworker into helping me out. O/w I can drop 2.6 from the matrix.

Please let me know which you would prefer.

**Comment by iHiD** [comment]
@BiggerNoise Thanks!
@dougal Tagging you here too as I'm not using Propono in production anywhere atm.
